src/viam/sdk/module/service.hpp

```diff
@@ -41,36 +80,18 @@ class ModuleService_ : public viam::module::v1::ModuleService::Service {
                                   const ::viam::module::v1::ValidateConfigRequest* request,
                                   ::viam::module::v1::ValidateConfigResponse* response) override;
 
-    ModuleService_(std::string addr);
-    ~ModuleService_();
+    void add_model_from_registry_inlock_(API api, Model model, const std::lock_guard<std::mutex>&);
+    void add_api_from_registry_inlock_(API api, const std::lock_guard<std::mutex>& lock);
+    Dependencies get_dependencies(google::protobuf::RepeatedPtrField<std::string> const& proto,
+                                  std::string const& resource_name);
+    std::shared_ptr<Resource> get_parent_resource(Name name);
 
-   private:
-    void add_model_from_registry_inlock_(std::shared_ptr<Server> server,
-                                         API api,
-                                         Model model,
-                                         const std::lock_guard<std::mutex>&);
-    void add_api_from_registry_inlock_(std::shared_ptr<Server> server,
-                                       API api,
-                                       const std::lock_guard<std::mutex>& lock);
     std::mutex lock_;
     std::shared_ptr<Module> module_;
     std::shared_ptr<RobotClient> parent_;
     std::string parent_addr_;
-};
-
-// C++ modules must handle SIGINT and SIGTERM. Make sure to create a sigset
-// for SIGINT and SIGTERM that can be later awaited in a thread that cleanly
-// shuts down your module. pthread_sigmask should be called near the start of
-// main so that later threads inherit the mask. SignalManager can handle the
-// boilerplate of this functionality.
-class SignalManager {
-   public:
-    SignalManager();
-    // wait on sigset
-    int wait(int* sig);
-
-   private:
-    sigset_t sigset;
+    std::shared_ptr<Server> server_;
+    std::unique_ptr<SignalManager> signal_manager_;
```

Comment:
> Seems like signal_manager_ could just be held by value? Why does it need to be in a separate allocation?

It can be and done!

> I'm also not sure why Server can't be a unique_ptr, ditto Module.

Ah `module_` can be a unique_ptr (I thought it couldn't for some reason); changed. As for `server_`, from my other comment: